packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/camera.dart

```diff
@@ -155,49 +147,131 @@ class CameraUpdate {
   /// the camera's target to a geographical location that is 50 to the east and
   /// 75 to the south of the current location, measured in screen coordinates.
   static CameraUpdate scrollBy(double dx, double dy) {
-    return CameraUpdate._(
-      <Object>['scrollBy', dx, dy],
-    );
+    return ScrollBy(dx, dy);
   }
 
   /// Returns a camera update that modifies the camera zoom level by the
   /// specified amount. The optional [focus] is a screen point whose underlying
   /// geographical location should be invariant, if possible, by the movement.
   static CameraUpdate zoomBy(double amount, [Offset? focus]) {
-    if (focus == null) {
-      return CameraUpdate._(<Object>['zoomBy', amount]);
-    } else {
-      return CameraUpdate._(<Object>[
-        'zoomBy',
-        amount,
-        <double>[focus.dx, focus.dy],
-      ]);
-    }
+    return ZoomBy(amount, focus);
   }
 
   /// Returns a camera update that zooms the camera in, bringing the camera
   /// closer to the surface of the Earth.
   ///
   /// Equivalent to the result of calling `zoomBy(1.0)`.
   static CameraUpdate zoomIn() {
-    return const CameraUpdate._(<Object>['zoomIn']);
+    return const ZoomIn();
   }
 
   /// Returns a camera update that zooms the camera out, bringing the camera
   /// further away from the surface of the Earth.
   ///
   /// Equivalent to the result of calling `zoomBy(-1.0)`.
   static CameraUpdate zoomOut() {
-    return const CameraUpdate._(<Object>['zoomOut']);
+    return const ZoomOut();
   }
 
   /// Returns a camera update that sets the camera zoom level.
   static CameraUpdate zoomTo(double zoom) {
-    return CameraUpdate._(<Object>['zoomTo', zoom]);
+    return ZoomTo(zoom);
   }
 
-  final Object _json;
-
   /// Converts this object to something serializable in JSON.
-  Object toJson() => _json;
+  Object toJson();// => _json;
+}
+
+/// Defines a camera move to a new position.
+class NewCameraPosition extends CameraUpdate {
```

Comment:
I think it would probably be easier to keep track of all of these classes if they had a `CameraUpdate` prefix.